### ISSUE TITLE
Cannot pull pod infra container image from a private repository on windows node

### ISSUE THREAD

**Issue by ylhyh** [kubernetes/kubernetes#76077]
**What happened**:
There is a Windows minion node in my k8s cluster, I have specified "--pod-infra-container-image" for kubelet which is pointing to a private docker repository. The kubelet running as a Windows service, service account is LocalSystem:
```
C:\Users\Administrator>sc qc kubelet
[SC] QueryServiceConfig SUCCESS

SERVICE_NAME: kubelet
        TYPE               : 10  WIN32_OWN_PROCESS
        START_TYPE         : 2   AUTO_START
        ERROR_CONTROL      : 0   IGNORE
        BINARY_PATH_NAME   : D:/k8s/kubelet.exe --kubeconfig=D:/k8s/pki/admin.conf --node-ip=172.16.126.216 --hostname-override=node009 --pod-infra-container-image=repo.q1lan.k8s:9999/q1winpause:0.1 --network-plugin=cni --cni-bin-dir=D:/k8s/cni/bin --cni-conf-dir=D:/k8s/cni/net.d --cert-dir=D:/k8s/kubelet/pki --volume-plugin-dir=D:/k8s/kubelet-plugins/volume/exec/ --log-dir=D:/k8s/log --log-file=D:/k8s/log/kubelet.log --logtostderr=false --config=D:/k8s/kubelet/config.yaml --resolv-conf="D:/k8s/resolv.conf" --windows-service
        LOAD_ORDER_GROUP   :
        TAG                : 0
        DISPLAY_NAME       : Kubernetes Node Agent
        DEPENDENCIES       : docker
                           : ovn-kubernetes
        SERVICE_START_NAME : LocalSystem
```
I have login to the repository with Administrator, and then copied config.json from Administrator's profile folder to "C:\windows\system32\config\systemprofile\\.docker".

When I schedule pod on Windows node, kubelet cannot pull infra container image from the private repository with error:
```
...
  Warning  FailedCreatePodSandBox  4m7s (x88 over 24m)  kubelet, node009   Failed create pod sandbox: rpc error: code = Unknown desc = failed pulling image "repo.q1lan.k8s:9999/q1winpause:0.1": Error response from daemon: Get http://repo.q1lan.k8s:9999/v2/q1winpause/manifests/0.1: no basic auth credentials
...
```

I can see there is a PR https://github.com/kubernetes/kubernetes/pull/46233 solved this issue, but it seems not work on Windows.

**What you expected to happen**:
Able to pull infra container image from a private repository on Windows node.

**How to reproduce it (as minimally and precisely as possible)**:
Run kubelet as a Windows service and specify --pod-infra-container-image to point to a private repository with basic auth.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):

Linux node:
```
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.3", GitCommit:"721bfa751924da8d1680787490c54b9179b1fed0", GitTreeState:"clean", BuildDate:"2019-02-01T20:08:12Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.3", GitCommit:"721bfa751924da8d1680787490c54b9179b1fed0", GitTreeState:"clean", BuildDate:"2019-02-01T20:00:57Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
```
Windows node:
```
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.3", GitCommit:"721bfa751924da8d1680787490c54b9179b1fed0", GitTreeState:"clean", BuildDate:"2019-02-01T20:08:12Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"windows/amd64"}
```
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`): Windows Server core 1709
- Kernel (e.g. `uname -a`): 10.0.16299.125
- Install tools:
- Others:

**Comment by ylhyh** [comment]
@kubernetes/sig-windows-bugs

**Comment by andyzhangx** [comment]
@ylhyh Your private repo only provides http service? does it need authentication?
 > `http://repo.q1lan.k8s:9999`

**Comment by ylhyh** [comment]
the private repo built by Nexus, it requires basic authentication

**Comment by bclau** [comment]
Hello,

First of all, the PR https://github.com/kubernetes/kubernetes/pull/46233 you mentioned was closed, but not merged.

Secondly, why do you need to use a private repo for the ``--pod-infra-container-image`` pause image? The general recommendation is to have the image prepulled on your node, or build / tag it there: [1], or use the official Microsoft image ``mcr.microsoft.com/k8s/core/pause:1.0.0`` [2]. Even if you opt to use a private repo, if you pull it after you've authenticated, it would fix your problem.

Finally, I can confirm that kubelet is not able to pull the configured ``--pod-infra-container-image`` if it is private. I've tried this on a Kubernetes v1.14.1 cluster. Note that creating pods using regular private images using Kubernetes secrets is a behaviour tested by one of the tests [3], and it is known to pass on the Windows CIs (but the test does not cover the ``--pod-infra-container-image`` private image case).

[1] https://docs.microsoft.com/en-us/virtualization/windowscontainers/kubernetes/joining-windows-workers?tabs=ManagementIP#tag-the-image
[2] https://github.com/microsoft/SDN/blob/master/Kubernetes/flannel/start-kubelet.ps1#L31
[3] https://github.com/kubernetes/kubernetes/blob/master/test/e2e/common/runtime.go#L398

**Comment by bclau** [comment]
Ok, so, I have a couple of updates regarding this.

First of all, indeed, the PR mentioned in the issue was closed without merging, but another one was merged in its stead: [1]

Secondly, it seems that kubelet is trying multiple credential providers, including for azure, aws, gcp, kubernetes secrets, and local configurations, as it can be seen here: [2].

I actually ended up getting it to work by setting up the local configrations properly. From what I can see in the code, it could still be a Windows related issue [3] (the env var name should be ``HOMEPATH`` on Windows, not ``HOME``), but it can be avoided, since it will check multiple locations for the configurations:

    I0513 16:22:05.542905   12204 config.go:131] looking for config.json at C:\var\lib\kubelet\config.json
    I0513 16:22:05.542905   12204 config.go:131] looking for config.json at C:\k\config.json
    I0513 16:22:05.542905   12204 config.go:131] looking for config.json at C:\k\.docker\config.json
    I0513 16:22:05.543892   12204 config.go:131] looking for config.json at C:\.docker\config.json
    I0513 16:22:05.543892   12204 config.go:101] looking for .dockercfg at C:\var\lib\kubelet\.dockercfg
    I0513 16:22:05.543892   12204 config.go:101] looking for .dockercfg at C:\k\.dockercfg
    I0513 16:22:05.543892   12204 config.go:101] looking for .dockercfg at C:\k\.dockercfg
    I0513 16:22:05.543892   12204 config.go:101] looking for .dockercfg at C:\.dockercfg
    I0513 16:22:05.543892   12204 provider.go:86] Unable to parse Docker config file: couldn't find valid .dockercfg after checking in [/var/lib/kubelet   /]

So, basically, I just had to copy the ``~/.docker/config.json`` file to one of those locations.

[1] https://github.com/kubernetes/kubernetes/pull/51870
[2] https://github.com/kubernetes/kubernetes/tree/master/pkg/credentialprovider
[3] https://github.com/kubernetes/kubernetes/blob/master/pkg/credentialprovider/config.go#L63

**Comment by ylhyh** [comment]
Yes, %HOMEDRIVE%+%HOMEPATH%  is the right profile path:
HOMEDRIVE=C:
HOMEPATH=\Users\Administrator

**Comment by bclau** [comment]
Hm, keep in mind that you're trying to run the Kubelet as a service, under the ``LocalSystem`` account, which has a totally different ``HOMEPATH``, which should be something like ``C:\Windows\System32\Config\systemprofile`` (probably).

**Comment by ylhyh** [comment]
@bclau sure, just to show the name of environment variable

**Comment by PatrickLang** [comment]
this has a PR open, so adding
/milestone v1.15
